scripts/publish-ci.ts

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env zx
+
+import { readFileSync } from 'node:fs'
+import { fileURLToPath } from 'node:url'
+import { $ } from 'zx'
+
+let version = process.argv[2]
+
+if (!version)
+  throw new Error('No tag specified')
+
+if (version.startsWith('v'))
+  version = version.slice(1)
+
+const pkgPath = fileURLToPath(new URL('../package.json', import.meta.url))
+const pkg = JSON.parse(readFileSync(pkgPath, 'utf-8'))
+
+if (pkg.version !== version) {
+  throw new Error(
+    `Package version from tag "${version}" mismatches with the current version "${pkg.version}"`,
+  )
+}
+
+const releaseTag = version.includes('beta')
+  ? 'beta'
+  : version.includes('alpha')
+    ? 'alpha'
+    : undefined
+
+console.log('Publishing version', version, 'with tag', releaseTag)
+
+if (releaseTag)
+  await $`pnpm -r publish --access public --tag ${releaseTag}`
```

Comment:
How about the `--provenance` flag? Looks like it could be added to `.npmrc` directly:

https://github.com/pnpm/pnpm/issues/6435#issuecomment-1518397267

Reply:
Oh, yes, right! I forgot about it